plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -608,8 +608,9 @@ private boolean onlyOnePluginIncluded() {
                     top.put("core", new JSONObject().accumulate("name", "core").accumulate("version", m.group(1)).accumulate("url", "https://foobar"));
                 }
 
-                //TODO: should it also scan detached plugins info?
-                m = Pattern.compile("WEB-INF/(?:optional-)?plugins/([^/.]+)[.][hj]pi").matcher(name);
+                // We should also scan detached plugins to make sure we get the proper groupId for detached not using the
+                // default org.jenkins-ci one
+                m = Pattern.compile("WEB-INF/(?:(optional-|detached-))?plugins/([^/.]+)[.][hj]pi").matcher(name);
```

Comment:
I guess the `-` char could be moved as below:
```suggestion
                m = Pattern.compile("WEB-INF/(?:(optional|detached)-)?plugins/([^/.]+)[.][hj]pi").matcher(name);
```

Reply:
By the way, is `-` character a special character or only when between `[]`?